### PR TITLE
Conditional ms build tasks

### PR DESCRIPTION
**Pull request by pi3k14**
Did a retake on this pull request - allows the user to conditionally disable GitVersions MSBuild tasks

Default behavior is to run all task (as before)

### CODE REVIEW

File: src/GitVersionTask/NugetAssets/GitVersionTask.targets [hunk 1]
```diff
@@ -4,31 +4,39 @@
...
-      NoFetch="$(GitVersion_NoFetchEnabled)"
-        />
+  <Target Name="WriteVersionInfoToBuildLog" BeforeTargets="CoreCompile" Condition="$(WriteVersionInfoToBuildLog) == 'true'">
+    <WriteVersionInfoToBuildLog SolutionDirectory="$(SolutionDir)" />
+  </Target>
+  
+  <Target Name="UpdateAssemblyInfo" BeforeTargets="CoreCompile" Condition="$(UpdateAssemblyInfo) == 'true'">
     <UpdateAssemblyInfo
-      SolutionDirectory="$(SolutionDir)"
-      NoFetch="$(GitVersion_NoFetchEnabled)"
```
Comment (pascalberger) [inline src/GitVersionTask/NugetAssets/GitVersionTask.targets]: Have you removed NoFetch intentionally? This was introduced to fix #646

File: src/GitVersionTask/NugetAssets/GitVersionTask.targets [hunk 2]
```diff
@@ -4,31 +4,39 @@
...
     AssemblyFile="$(MSBuildThisFileDirectory)..\..\GitVersionTask.dll"  />
   <UsingTask
       TaskName="GitVersionTask.WriteVersionInfoToBuildLog"
       AssemblyFile="$(MSBuildThisFileDirectory)..\..\GitVersionTask.dll" />
 
-  <Target Name="UpdateAssemblyInfo"
-          BeforeTargets="CoreCompile">
-    <WriteVersionInfoToBuildLog
-      SolutionDirectory="$(SolutionDir)"
-      NoFetch="$(GitVersion_NoFetchEnabled)"
```
Comment (pascalberger) [inline src/GitVersionTask/NugetAssets/GitVersionTask.targets]: Have you removed NoFetch intentionally? This was introduced to fix #646

File: src/GitVersionTask/NugetAssets/GitVersionTask.targets (hunk 1 above)
Comment (pi3k14) [inline src/GitVersionTask/NugetAssets/GitVersionTask.targets]: No - it's a diff I didn't catch when creating a patch from my "messy" repo.

**Comment by pi3k14** [comment]
Added NoFetch back (and if someone wonder, I am not going to touch git -rebase to make this any "prettier" ;)

File: src/GitVersionTask/NugetAssets/GitVersionTask.targets (hunk 2 above)
Comment (pi3k14) [inline src/GitVersionTask/NugetAssets/GitVersionTask.targets]: An oversight - fixed

**Comment by pascalberger** [comment]
LGTM. The only thing I'm not sure is, that with now three individual targets, all dependent on the same target, is it ensured that they are always called in the correct order?

**Comment by pi3k14** [comment]
I didn't think there was a dependency between this targets, if so this PR breaks up. The intention was to switch of the tasks you didn't want (UpdateAssemblyInfo in my case, because I want to handle AssemblyVersion, AssemblyFileVersion and AssemblyInformationalVersion individually, and so far I haven't had any problems with it).

This solution only assure that the tasks are run before CoreCompile, in no particular order.

If there is a dependency between them, I can't switch them off at will. Do you have any further information about this?

**Comment by pascalberger** [comment]
@JakeGinnivan Do you know if the execution order is an issue here?

**Comment by JakeGinnivan** [comment]
Not sure.. @SimonCropp ?

**Comment by pi3k14** [comment]
The original order was; WriteVersionInfoToBuildLog, UpdateAssemblyInfo and GetVersion.

WriteVersionInfoToBuildLog cant be dependent on any of the other task. 
I have run a great deal of builds with only GetVersion, and it works without any problems (also tested on new 1.89 build agent on TFS, RC1 version).

The remaining questions then are - is UpdateAssemblyInfo dependent on WriteVersionInfoToBuild Log, would running WriteVersionInfoToBuildLog after the other tasks creating havoc?
To me, that would make no sense.

**Comment by JakeGinnivan** [comment]
This is an area of GitVersion I don't know very well so it all looks fine to me. @pascalberger if you can't see anything, lets merge

**Pull request merged**